### ISSUE TITLE
Error in java_srvs.py, jar generation

### ISSUE THREAD

**Issue by chadrockey** [rosjava/rosjava_core#21]
_Original author: icodera...@gmail.com (July 07, 2011 22:25:33)_

Build fails with missing AddTwoInts, Chars, Int64s etc.

Reason:

a. JAR files have absolute paths, not relative ones

b. Class generation in java_srvs.py is incorrect

Instead of 
- SERVICE_CLASS = 'org.ros.message.Service'
  it should be
- SERVICE_CLASS = 'org.ros.service.Service'

Additional Info

Fresh checkout of rosjava_core

_Original issue: http://code.google.com/p/rosjava/issues/detail?id=21_

**Comment by chadrockey** [comment]
_From icodera...@gmail.com on July 12, 2011 15:30:06_
A better look into things leads me to the conclusion that the python script may not be at fault in this respect, and that it does not need to be modified. However I had to modify a few source files.

The issue that occurs is that during compilation of the roscpp.jar file, issues are generated. Something like this

  [javac] home/kshaurya/.ros/rosjava/gen/srv/roscpp/org/ros/service/roscpp/Empty.java:4: package org.ros.internal.message does not exist
    [javac] import org.ros.internal.message.Service;

Eventually I realized that the test_ros messages were not being generated, and so I had to manually generate the message and service files to ~/.ros/rosjava using the python scripts, change their path to point to /~.ros/rosjaava/gen/org.....

AND modify the three roscpp files (Empty, GetLoggers, SetLoggerLevel) to import org.ros.message.Service rather than org.ros.service.Service.

Hope this helps in providing a better picture.
